Fixes to RE

mikecappella opened the pull request:
RE fixes to avoid unintended Perl var $, and add # to the RE sequence to catch anchors.  Remove redundant a-z (caught w/ i modifier), and replace 0-9 with /d.  The char . does not need escaping, nor does dash when it is at the front of the char class.

mikecappella committed:
Fixes to RE

RE fixes to avoid unintended Perl var $, and add # to the RE sequence to catch anchors.  Remove redundant a-z (caught w/ i modifier), and replace 0-9 with /d.  The char . does not need escaping, nor does dash when it is at the front of the char class.

michaelherger commented on Slim/Web/Template/.SkinManager.pm.swp:
Please get rid of this file.

mikecappella replied:
Sorry about that.  GitHub Desktop + git frustrates me beyond belief with endless threads of incorrect or obsolete documentation.  I think I've done the correct thing now.

michaelherger commented on Slim/Web/Template/SkinManager.pm:
```diff
@@ -263,9 +263,9 @@ sub _parseURIs {
 	return $text unless $text;
 
-	if (!($text =~ s!\b(https?://[A-Za-z0-9\-_\.\!~*'();/?:@%&=+$,]+)!<a href=\"$1\" target=\"_blank\" class="link">$1</a>!igo)) {
+	if (!($text =~ s!\b(https?://[-A-Z\d_.~*'();/?:@%&=+,#\$\!]+)!<a href=\"$1\" target=\"_blank\" class="link">$1</a>!igo)) {
```
Does moving the exclamation point have any significance?

mikecappella replied:
No, it was simply to group all the escaped items at the end of the RE.

mikecappella committed:
Create .gitignore

mikecappella committed:
Fix RE

RE fixes to avoid unintended Perl var $, and add # to the RE sequence to catch anchors.  Remove redundant a-z (caught w/ i modifier), and replace 0-9 with /d.  The char . does not need escaping, nor does dash when it is at the front of the char class.

mikecappella committed:
Merge branch 'public/9.1' of https://github.com/mikecappella/slimserver into public/9.1

mikecappella committed:
remove .SkinManager.pm.swp

mikecappella committed:
remove .gitignore

michaelherger commented:
Thanks!
